serialization_v2/serialization.py

```diff
@@ -0,0 +1,1280 @@
+                    f"cannot handle {value!r} (type {type(value).__name__}) as global,"
+                    f" qualname {qualname} not found,"
+                    f" no {'.'.join(qualname_parts[: i + 1])} in module {mod_name}"
+                )
+            obj.append(getattr(obj[-1], qualname_parts[i]))
+        if obj[-1] is not value:
+            raise SerializationError(
+                f"cannot handle {value!r} (type {type(value).__name__}) as global,"
+                f" qualname {qualname} gives different object {obj[-1]!r}"
+            )
+        if len(qualname_parts) > 1:
+            base_obj_repr = self._serialize_value(obj[-2], prefix=name + "_base")
+            return PyEvalCode(f"{base_obj_repr.py_inline()}.{qualname_parts[-1]}")
+        if "." in mod_name:
+            # Maybe we can shorten the import.
+            # Check if some of the parent modules already import the object.
+            mod_name_parts = mod_name.split(".")
+            for i in range(len(mod_name_parts)):
+                parent_mod_name = ".".join(mod_name_parts[: i + 1])
+                mod = sys.modules.get(parent_mod_name)
+                if mod and getattr(mod, qualname, None) is value:
+                    mod_name = parent_mod_name  # we can directly use this
+                    break
+        self._setup_module_import(mod_name)
+        return PyCode(
+            py_name=name,
+            value=value,
+            py_code=f"from {mod_name} import {qualname}\n"
+            if qualname == name
+            else f"from {mod_name} import {qualname} as {name}\n",
+        )
+
+    def _serialize_module(self, value: ModuleType, name: str) -> PyCode:
+        mod_name = value.__name__
+        assert sys.modules[mod_name] is value
+        self._setup_module_import(mod_name)
+        if "." not in mod_name:
+            return PyCode(
+                py_name=name,
+                value=value,
+                py_code=f"import {mod_name}\n" if mod_name == name else f"import {mod_name} as {name}\n",
+            )
+        mod_name, qualname = mod_name.rsplit(".", 1)
+        return PyCode(
+            py_name=name,
+            value=value,
+            py_code=f"from {mod_name} import {qualname}\n"
+            if qualname == name
+            else f"from {mod_name} import {qualname} as {name}\n",
+        )
+
+    def _setup_module_import(self, mod_name: str):
+        """make sure that the import works, by preparing ``sys.path`` if necessary"""
+        if "." in mod_name:
+            mod_name = mod_name.split(".", 1)[0]
+        if mod_name in self.known_modules:
+            return
+        mod = sys.modules[mod_name]
+        if not hasattr(mod, "__file__"):
+            return  # assume builtin module or so
+        mod_path = _get_module_path_from_module(mod)
+        self.add_sys_path(mod_path)
+        self.known_modules.add(mod_name)
+
+    def add_sys_path(self, path: str, *, recursive: bool = True):
```

Comment:
```suggestion
    def add_sys_path(self, path: str, *, recursive: bool = True, code_comment: str = ""):
```
And then below add that `code_comment` as a prefix to the `PyCode` `py_code` string.

Reply:
(This is an addition I made in the meantime after you made the PR.)